x/jwt_helper.go

```diff
@@ -20,17 +20,31 @@ import (
 	"context"
 	"fmt"
 
-	"github.com/dgrijalva/jwt-go"
+	"github.com/golang-jwt/jwt"
 	"github.com/pkg/errors"
 )
 
 func ParseJWT(jwtStr string) (jwt.MapClaims, error) {
 	token, err := jwt.Parse(jwtStr, func(token *jwt.Token) (interface{}, error) {
-		if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
-			return nil, errors.Errorf("unexpected signing method: %v",
-				token.Header["alg"])
+		if WorkerConfig.UsePublicKey {
+			if _, ok := token.Method.(*jwt.SigningMethodRSA); !ok {
+				return nil, errors.Errorf("unexpected signing method: %v",
+					token.Header["alg"])
+			}
+
+			key, err := jwt.ParseRSAPublicKeyFromPEM([]byte(WorkerConfig.HmacSecret))
```

Comment:
we should do this parsing just one, not for every JWT token.

Reply:
Also, I believe this could have different formats. I can find this out if needed. We should support or at least we should know what we do not support/support.